### DISCUSSION TITLE
Combine other phone sensor data

### DISCUSSION THREAD

**Discussion by stephend61** [j-kaltes/Juggluco#72] [Ideas]
Hi , this is a great app, it's made my life so much better keeping track of my bg levels.
I was thinking it might be good record other phone sensor data with the readings, eg. Heart rate, GPS location, just to help identify different things that impact your reading, the ability to add actual notes would be nice too.
I am currently mirroring the data to my pc where I plot the data using grafana, would be nice to plot heart rate and location along side it.
Thanks.

**Comment by j-kaltes** [comment]
Yes, heart rate could easily be added for WearOS Juggluco - or Garmin Kerfstok users. Also activities could be send from a Garmin watch to Juggluco.
Sometimes heart rate is used as the only measure of activity level and could be used to help predict glucose level and adjuct the insulin dose. Because I have multiple watches I am a little bit sceptical about these heart rate measurements: Samsung Galaxy Watch 4 gives sometimes a heart rate that is three times as large than Garmin Vivoactive 3.
I don't know what Juggluco could do with GPS data.
For what do you need this "_ability to add actual notes_"? In Juggluco one can enter amounts under different labels. Up to 40 different labels are possible. Is this not enough for your notes? Under the carb category you can also compose a meal out of its components. So you can specify meals in detail.

**Reply by stephend61** [reply]
Hi, The thinking with GPS location was it help me know what I was doing at the time of the read, especially if it was unusually high, as most of my higher readings these days are caused by activity over food.  I do make use of the current labels but notes would be better as labels don't tell you much unless you have a lot of them. 
Re the samsung watch I have found similar issues but most of the time it is close to the mark, I have been checking it against my chest strap heart rate monitor, when exercising is really when it can be way off and that's when I used the chest one.
Thanks.

**Comment by j-kaltes** [comment]
I don't think that Juggluco should keep track of peoples location. For what you want that is also not necesssary. To know where you was at a certain day and time, you can use the location logging of another app. See:
https://android.gadgethacks.com/how-to/android-basics-view-your-location-history-0168748/

**Reply by stephend61** [reply]
It's not like it's being sent anywhere except your mirror , if you set one. It's only for your own use .
I will see I can collector it by other means, I was hoping to plot it against my levels, not sure any of the other apps exported that data in a useful format if at all.
Thanks.

**Reply by stephend61** [reply]
I looked at the google location history but it's use here would be pretty limited as you have to manually download the data each day, as Google only give an option to get it every 2 months automatically.
Might have to see if i can write something to send to my pc.
